caddyconfig/httpcaddyfile/serveroptions.go

```diff
@@ -141,22 +140,51 @@ func unmarshalCaddyfileServerOptions(d *caddyfile.Dispenser) (any, error) {
 				}
 				serverOpts.ShouldLogCredentials = true
 
+			case "protocols":
+				protos := d.RemainingArgs()
+				for _, proto := range protos {
+					if proto != "h1" && proto != "h2" && proto != "h2c" && proto != "h3" {
```

Comment:
Small thing, in other languages I tend to use an "array includes" pattern when I'm checking if a particular variable is one of many values. In Go, this hasn't been easy because slices don't have per-type functions to do this in a typesafe way, but with generics, we have a way now:

```go
import "golang.org/x/exp/slices"
```

```suggestion
					if slices.Contains([]string{"h1", "h2", "h2c", "h3"}, proto) {
```

If you don't want to pull this in just for this, that's fine, but I think this reads cleaner.

Reply:
Oh, that's cool! I am not sure I'm ready for Caddy to use generics quite yet (I know they're one of Go's most thoroughly vetted features ever, I just want to wait a little longer), especially here where the gain is very minimal (and it adds a dependency for... string comparison) but I do see the value of this in general and I think when we are ready for generics, something like this would be a nice use case!